Sources/VaporPostgreSQL/Provider.swift

```diff
@@ -1,34 +1,145 @@
+import URI
 import Vapor
+import Fluent
 import FluentPostgreSQL
 
+public typealias PostgreSQLDriver = FluentPostgreSQL.PostgreSQLDriver
+
 public final class Provider: Vapor.Provider {
-    /*
-        PostgreSQL database driver created by the provider 
+    public let provided: Providable
+
+    public enum Error: Swift.Error {
+        case noPostgreSQLConfig
+        case missingConfig(String)
+    }
+
+    /**
+        PostgreSQL database driver created by the provider.
     */
     public let driver: PostgreSQLDriver
 
-    /*
-        PostgreSQL database created by the provider
+    /**
+        Creates a PostgreSQL provider from a `postgresql.json` config file.
+
+        The file should contain similar JSON:
+
+            {
+                "host": "127.0.0.1",
+                "user": "postgres",
+                "password": "",
+                "database": "test",
+                "port": 5432, // optional
+            }
+
+        Optionally include a url instead:
+
+            {
+                "url": "psql://user:pass@host:5432/database"
+            }
+
     */
-    public let database: DatabaseDriver?
-
-    /*
-        Initialize PostgreSQL database running on host
-        - parameter host: May be eitehr  a host name or an IP address. default localhost 
-        - parameter port: If port is not 0 the value is used as the port number for the TCP/IP connection. default 5432
-        - parameter dbname: Name of database
-        - parameter user: PostgreSQL login ID
-        - parameter password: Password for user
-        - throws: Error.cannotEstablishConnection if the call to PostgreSQL fails. Note this attempts to execute a SELECT version() to the db to make sure there is a connection
+    public convenience init(config: Config) throws {
+        guard let postgresql = config["postgresql"]?.object else {
+            throw Error.noPostgreSQLConfig
+        }
+
+        if let url = postgresql["url"]?.string {
+            try self.init(url: url)
+        } else {
+            guard let host = postgresql["host"]?.string else {
+                throw Error.missingConfig("host")
+            }
+
+            guard let user = postgresql["user"]?.string else {
+                throw Error.missingConfig("user")
+            }
+
+            guard let password = postgresql["password"]?.string else {
+                throw Error.missingConfig("password")
+            }
+
+            guard let dbname = postgresql["database"]?.string else {
+                throw Error.missingConfig("database")
+            }
+
+            let port = postgresql["port"]?.int
+
+            try self.init(
+                host: host,
+                port: port!,
```

Comment:
But if you force unwrap here it will crash if no port exists in the config files. You'll need to allow an optional in the init call then use the default port if nil is passed.